internal/datadog/go.mod

```diff
@@ -0,0 +1,72 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog
+
+go 1.19
+
+require (
+	github.com/DataDog/datadog-agent/pkg/trace v0.48.0-devel
+	github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics v0.5.2
```

Comment:
If this gets merged after the PR @songy23 is working on we should bump these appropriately